jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/RuleSetNodeInstance.java

```diff
@@ -75,6 +75,8 @@
     private static final Logger logger = LoggerFactory.getLogger(RuleSetNodeInstance.class);
     
     private static final String ACT_AS_WAIT_STATE_PROPERTY = "org.jbpm.rule.task.waitstate";
+    private static final String FIRE_RULE_LIMIT_PARAMETER = "FireRuleLimit";
+    private static final int DEFAULT_FIRE_RULE_LIMIT = Integer.parseInt(System.getProperty("org.jbpm.rule.task.firelimit", "10000"));
```

Comment:
Are we sure that we want to enable this for everyone, always? My opinion is that this should default to disabled, for the sake of backwards compatibility - if it wasn't a problem for them until now, let's not have them think about this. Especially given the arbitrary nature of the threshold.

Reply:
I actually think it is better to enable it by default mainly due to number of people ran into looping rules and making their servers unresponsive (and thus had to kill them) just ti figure out they made mistake in the rule that simply gets into infinitive loop. SO better would be to get few failures due to limit instead of having issues with high cpu utilisation and trying to figure what's wrong with my server.

Reply:
Yeah, you're right. I forgot that this does actually kill the server if it happens.